clippy_lints/src/methods/suspicious_to_owned.rs

```diff
@@ -17,19 +17,25 @@ pub fn check(cx: &LateContext<'_>, expr: &hir::Expr<'_>, recv: &hir::Expr<'_>) -
         let input_type = cx.typeck_results().expr_ty(expr);
         if let ty::Adt(adt, _) = cx.typeck_results().expr_ty(expr).kind();
         if cx.tcx.is_diagnostic_item(sym::Cow, adt.did());
+
         then {
             let mut app = Applicability::MaybeIncorrect;
             let recv_snip = snippet_with_context(cx, recv.span, expr.span.ctxt(), "..", &mut app).0;
-            span_lint_and_sugg(
+            span_lint_and_then(
                 cx,
                 SUSPICIOUS_TO_OWNED,
                 expr.span,
                 &with_forced_trimmed_paths!(format!(
                     "this `to_owned` call clones the {input_type} itself and does not cause the {input_type} contents to become owned"
                 )),
-                "consider using, depending on intent",
-                format!("{recv_snip}.clone()` or `{recv_snip}.into_owned()"),
-                app,
+                |diag| {
+                    diag.span_suggestions(
+                        expr.span,
+                        "depending on intent, either make the Cow an Owned variant or clone the Cow itself",
+                        [format!("{recv_snip}.into_owned()"), format!("{recv_snip}.clone()")],
+                        Applicability::Unspecified
```

Comment:
Since this splits the combined suggestion into two overlapping suggestions this should keep the same applicability it had before.

Reply:
You're right, it should. Fixed it now!